Feature | db based on diagram

ITurres opened the pull request:
# 🚩We have made the following changes to complete Vet Clinic - Create db based on a diagram

## Here is a summary of what has been done

### Type of **`files`** changed/added

![My Skills](https://skillicons.dev/icons?i=postgresql,md), **PNG**

### In this new Branch, We have **➕ADDED➕** the succeeding **\*📂 folders** and/or **\*\*📄 files** to complete this milestone's [requirements](https://github.com/microverseinc/curriculum-databases/blob/main/db-structure/database_from_diagram.md)

# \*📂 root/

- **\*\*📄 schema_based_on_diagram** ➕

  - We have added the schema based on the diagram following the above requirements.

- **\*\*🖼️ clinic_diagram.png** ➕
    
      - We have added the diagram ERD.


### We have **✏️MODIFIED✏️** the succeeding **\*📂 folders** and/or **📄\*\* files**

# \*📂 root/

- **\*\*📄 README.md** ✏️

  - We have modified the documentation to add author 2 @mwismann.

- **\*\*📄 LICENSE** ✏️

  - We have added @mwismann.

---

### Thank you for taking the time to review this PR ⭐

---

If you require additional information or have any questions, please contact me on Slack as _**Arturo (Arthur) Emanuel Guerra Iturres**_, _**Mathías Wismann**_ We'll be happy to help you out. 🎯

---

ITurres committed:
Add 'schema_based_on_diagram.sql' and:

- Add -medical_histories- table.
- Add -patients- table.
- modify -medical_histories- table to add constraint to -patient_id- column.

ITurres committed:
docs: Modified 'README.md', 'LICENSE' and:

- Added 'clinic_diagram.png' to the repository.
- Updated 'README.md' to include the diagram,
  added author 2 data and update future features.
- Update 'LICENSE' to include author 2 name.

mwismann committed:
Add invoice and invoice_items tables

ITurres committed:
feat: on 'schema_based_on_diagram.sql' Added:

- Added -treatments- table.
- Added -treatments_history- join/junction table.

mwismann committed:
Add FK indexes for medical_histories, invoices and invoice_items

ITurres committed:
feat: on 'schema_based_on_diagram.sql' add:

- Add 'idx_treatments_history_medical_history_id' index to 'treatments_history' table.
- Add 'idx_treatments_history_treatment_id' index to 'treatments_history' table.

Felix45 commented on schema_based_on_diagram.sql:
```diff
@@ -0,0 +1,63 @@
+  total_price DECIMAL(10,2) NOT NULL,
+  invoice_id INT,
+  CONSTRAINT fk_invoice_id FOREIGN KEY (invoice_id) REFERENCES invoices(id) ON DELETE CASCADE,
+  treatment_id INT NOT NULL,
```
- [ Required ] I think it would be nice if you add a `foreign key` constraint to the `treatment_id` column since it is supposed to be a reference of column `id` in the `treatments` table in the `invoice_items` table. Once you've added the foreign key constraint kindly remember to add an `index` to the `treatment_id` column the same way you've done for other foreign keys this will speed up search queries where the column is involved.

## Example code
```SQL
ADD CONSTRAINT fk_treatment_id FOREIGN KEY (treatment_id) REFERENCES treatments(id)

CREATE INDEX idx_invoice_items_treatment_id ON invoice_items(treatment_id);
```

ITurres replied:
Hi @Felix45! thank you for reviewing this PR 🎯 and for such valuable feedback! we really appreciated it! 🤝 

Absolutely! It seems we have overlooked it 🥲, thank you for pointing it out!

Changes will be made to apply the suggested required changes.

Cheers, and happy coding! 🤠 

@ITurres @mwismann

Felix45 reviewed (changes requested):
Hi @ITurres & @mwismann, 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

<img src="https://user-images.githubusercontent.com/57562869/143511358-9cde7daf-4ad0-4db2-8f15-18b54288ff91.gif" width="350px" />

## Highlights ✔️ 
- [x] Created database according to the ERD diagram ✅ 
- [x] Some foreign keys have indexes ✅ 
- [x] Git flow is used correctly ✅  
- [x] Repository has a professional readme file ✅  
- [x] Pull request has a descriptive title and summary ✅   

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

You can also consider:

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

ITurres committed:
fix: on 'schema_based_on_diagram.sql' file:

- Added table alteration for -invoice_items- table
 to add foreign key constraint.
- Added new index for -invoice_items- table
 to improve performance.
- Renamed foreign key constraint on -treatments_history- table.

Strangeal reviewed (approved):
Hi @ITurres, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

![Alt Text](https://media.giphy.com/media/JoSiL3L4ON8DlyVxv0/giphy.gif)

## Highlights
- Added foreign key constraint ✅

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

ITurres commented:
Hi @Strangeal Justice! cheers for the approval! 💯 Wish you guys a great week!
